### PR TITLE
[omxplayer] A few omxplayer fixes

### PR DESCRIPTION
**Pull request by popcornmix**
A few omxplayer issues emerged with Isengard test builds.

### CODE REVIEW

**Comment by popcornmix** [comment]
jenkins build this please

**Comment by popcornmix** [comment]
@FernetMenta can you okay the first commit in particular as that edits dvdplayer. It should be pretty harmless.

**Comment by FernetMenta** [comment]
I don't claim that I understand all this code but the part in dvdplayer is easier to read now. Good work.

File: xbmc/cores/omxplayer/OMXHelper.cpp [hunk 1]
```diff
@@ -44,6 +44,9 @@ static bool PredicateVideoPriority(const SelectionStream& lh, const SelectionStr
 
 bool OMXPlayerUnsuitable(bool m_HasVideo, bool m_HasAudio, CDVDDemux* m_pDemuxer, CDVDInputStream* m_pInputStream, CSelectionStreams &m_SelectionStreams)
 {
+  // if no OMXPlayer acceleration then it is not suitable
+  if (!CSettings::Get().GetBool("videoplayer.useomxplayer"))
```
Comment (anaconda) [inline xbmc/cores/omxplayer/OMXHelper.cpp]: I guess this is supposed to be `CSettings::SETTING_VIDEOPLAYER_USEOMXPLAYER` after #7721:
https://github.com/xbmc/xbmc/blob/master/xbmc/settings/Settings.cpp#L171

(While at it, `CSettings::SETTING_VIDEOPLAYER_USEMMAL` a couple lines below.)

**Comment by popcornmix** [comment]
Fixed to use new CSettings constants (thanks @anaconda).
jenkins build this please

**Pull request merged**